src/ControlSystem/Trigger.hpp

```diff
@@ -0,0 +1,142 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <limits>
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+
+#include "ControlSystem/Metafunctions.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/FunctionsOfTimeAreReady.hpp"
+#include "Evolution/EventsAndDenseTriggers/DenseTrigger.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace Tags {
+struct Time;
+}  // namespace Tags
+namespace control_system::Tags {
+struct MeasurementTimescales;
+}  // namespace control_system::Tags
+/// \endcond
+
+namespace control_system {
+/// \ingroup ControlSystemsGroup
+/// \ingroup EventsAndTriggersGroup
+/// Trigger for control system measurements.
+///
+/// This trigger is only intended to be used with the
+/// `control_system::Event` event.  A specialization of this trigger
+/// will be created during control system initialization for each \ref
+/// control_system::protocols::Measurement "measurement".
```

Comment:
Maybe add "for each unique measurement" to clarify that even though multiple control systems may use the same measurement, each measurement will only be triggered by one trigger.

Reply:
I have also updated the text in Event.hpp to match.